src/main/java/org/jboss/jandex/MethodInfo.java

```diff
@@ -141,14 +141,46 @@ public final String name() {
     }
 
     /**
-     * Returns the name of the given parameter.
+     * Returns the name of the given parameter, omitting synthetic parameters.
      * @param i the parameter index
-     * @return the name of the given parameter, or null.
+     * @return the name of the given parameter, omitting synthetic parameters, or null.
      */
     public final String parameterName(int i) {
         return methodInternal.parameterName(i);
     }
+
+    /**
+     * Returns the type of the given parameter, omitting synthetic parameters. This
+     * does not support local classes.
+     * @param i the parameter index
+     * @return the type of the given parameter, omitting synthetic parameters, or null.
+     * @throws UnsupportedOperationException if this method's container class is a local class or an enum.
+     */
+    public Type realParameter(int parameter) {
```

Comment:
Not totally excited by the name of the method, given we have `parameterName()` just above, also omitting synthetic parameters.

Other than that, it definitely looks like what I was looking for when I asked for it.

Reply:
`parameterTypeSkippingSyntheticParams()`?

Reply:
How about an overloaded method simply called `parameterType` to allow default behavior of using only "natural" parameters and also a way to access the synthetic type without direct access to the collection?

```java
public Type parameterType(int parameter) {
    return parameterType(parameter, false);
}

public Type parameterType(int parameter, boolean allowSynthetic) {
    // Same logic from realParameter in PR, but considering allowSynthetic value
}
```